src/analyses/README.md

```diff
@@ -92,7 +92,64 @@ To be documented.
 
 \subsection analyses-dependence-graph Data- and control-dependence analysis (dependence_grapht)
 
-To be documented.
+### Dependence graph
+
+Implemented in `src/analyses/dependence_graph.h(cpp)`. It is a graph and an
+abstract interpreter in the same time. The abstract interpretation nature
+allows a dependence graph to build itself (the graph) from a given GOTO program.
+
+A dependence graph extends the class `grapht` with `dep_nodet` as the type of
+nodes (see `src/util/graph.h` for more details about
+[graph representation](../util/README.md)).
+The `dep_nodet` extends `graph_nodet<dep_edget>` by an iterator to a GOTO
+program instruction. It means that each graph node corresponds to a particular
+program instruction. A labelled edge `(u, v)` of a dependence graph expresses
+a dependency of the program instruction corresponding to node `u` on the program
+instruction corresponding to node `v`. The label of the edge (i.e. data of the
+type `dep_edget` attached to the edge) denotes the kind of dependency. It can be
+`control-dependency`, `data-dependency`, or both.  
+
+#### Control dependency
+
+An instruction `j` corresponding to node `v` is control-dependent on instruction
+`i` corresponding to node `u` if and only if `j` post-dominates at least one
+but not all successors instructions of `i`.
+
+An instruction `j` post-dominates an instruction `i` if and only if each
+execution path going through `i` eventually reaches `j`. 
+
+Post-dominators analysis is implemented in `src/analyses/cfg_dominators.h(cpp)`.
+
+#### Data dependency
+
+The instruction `j` corresponding to node `v` is data-dependent on the
+instruction `i` corresponding to node `u` if and only if `j` may read data from
+memory location defined (i.e. written) by `i`.
+
+Reaching definitions analysis together with read-write ranges analysis are used
+to check whether one instruction may read data writen by another instruction.
+For more details see `src/analyses/reaching_definitions.h(cpp)` and
+`src/analyses/goto_rw.h(cpp)`.
+
+#### Construction
+
+The dependence graph extends the standard abstract interpreter class `ait`
+with post-dominators analysis and reaching definitions analysis. The domain of
+the abstract interpreter is defined in the class `dep_graph_domaint`.
+
+For each instruction `i` there is created an instance of `dep_graph_domaint`
+associated with `i`. The instance stores a set `control_deps` of program
+instructions the instruction `i` depends on via control-dependency, and a set
+`data_deps` of program instructions the instruction `i` depends on via
+data-dependency. These sets are updated (increased) during the computation,
```

Comment:
I feel like this last sentence is too complicated, but I can't figure out how to reword it. Does anyone have a good suggestion for this?

Reply:
I think it would be good to make clear we are talking about an abstract domain, and we have domain elements. Each domain element is made up of sets of control- and data dependencies (`control_deps` and `data_deps`, respectively).